### ISSUE TITLE
OpsAgent should sanitize GCE instance metadata before inserting them to otelopscol configuration file

### ISSUE THREAD

**Issue by ostash** [GoogleCloudPlatform/ops-agent#1221]
**Describe the bug**
OpsAgent generates a configuration file for otelopscol. For `prometheus` receiver it adds all GCE instance metadata as `__meta_*` labels, i.e.:

```
  prometheus/prometheus:
    config:
      global:
        scrape_interval: 1m
        scrape_timeout: 10s
        evaluation_interval: 1m
      scrape_configs:
      - job_name: hps_scraping
        honor_timestamps: true
        scrape_interval: 30s
        scrape_timeout: 10s
        metrics_path: /metrics
        scheme: http
        follow_redirects: true
        enable_http2: true
        static_configs:
        - targets:
          - localhost:5099
          labels:
            __meta_gce_instance_id: "...."
            __meta_gce_instance_name: ...
            __meta_gce_interface_ipv4_nic0: .....
            __meta_gce_machine_type: projects/..../machineTypes/n2-standard-2
            __meta_gce_metadata_key1: "value1"
            __meta_gce_metadata_key2: "value2"
```

If the metadata value contains something like "${foo:bar}" it will be copied as is to otelopscol config file

```
          labels:
            __meta_gce_metadata_key1: "some text ${foo:bar} more text"
```

This will be processed by otelopscol as a URI. In a simple case, this will cause configuration parsing failure (scheme "foo" is not supported for URI "foo:bar").

However, this can be used for injecting valid URI like ${file:path} or ${http://adress}. This can be potentially used for exposing sensitive data.

**To Reproduce**
1. Start a GCE VM with some Google provided image 
2. Ensure OpsAgent is installed
3. Add metadata to the VM with arbitrary key and value like: "${foo:bar}"
4. Customize the agent with configuration like below and restart the agent:
```
metrics:
  receivers:
    prometheus:
        type: prometheus
        config:
          scrape_configs:
            - job_name: 'job'
              scrape_interval: 30s
              metrics_path: /metics
              static_configs:
                - targets: ['localhost:5099']
```
5. Restart OpsAgent
5. See an error in the system logs

**Expected behavior**
GCE Instance Metadata must be sanitized.

**Environment (please complete the following information):**
 - Project ID: private
 - VM ID: private
 - VM distro / OS: RHEL7
 - Ops Agent version: google-cloud-ops-agent-2.29.0-1.el7.x86_64
 - Ops Agent configuration: see above

**Additional context**
This really looks like a security issue. The problem also can be quite easily triggered by the shell startup script provided in metadata, perfectly valid bash `var1=${var2:0:-2}` will break metrics collection.

**Comment by ostash** [comment]
Case 44829496

**Comment by jefferbrecht** [comment]
This is now fixed in Ops Agent 2.32.0.
